src/viewer.js

```diff
@@ -3957,23 +3960,38 @@ function doViewerResize(viewer, containerSize){
     THIS[viewer.hash].forceResize = false;
 }
 
-function handleArrowKeys(viewer) {
+function handleNavKeys(viewer) {
+
+    // Iterate over all keys to handle keyboard-based panning.
+    // If a key is held down or virtually held, increment its pan distance by panStep.
+    // Once the minimum pan distance is reached, stop virtually holding the key and reset the counter.
+
+    for (let code in viewer._navKeysDown) {
+        if (viewer._navKeysDown[code] || viewer._navKeysVirtuallyHeld[code]) {
+            // pan by step
+            viewer._navKeyFrames[code]++;
+            if (viewer._navKeyFrames[code] >= viewer._minNavKeyFrames) {
+                viewer._navKeysVirtuallyHeld[code] = false;
+                viewer._navKeyFrames[code] = 0;
```

Comment:
We shouldn't be resetting the _navKeyFrames here. On the one hand it's unnecessary (since it gets reset when the key gets pressed) but also it's problematic (because it resets the count if the key is still held down).